Sheepish-Squirrels/bangazon-api-sprint1
# Added GET, POST, PUT, and DELETE functionality to products

Fixes #2

Changes proposed in this pull request:

- npm installs:
body-parser
- set up function to get all products, route to see all products, added scripts in package.json to start, test, reset data, and generate data
- wrote function to get one product using route params
- changed file paths to be in appropriately labeled folders
- added body-parser to package.json to see body method on req, which was applied as middleware to app.js. Also, added post one product function
- used PUT to edit products
- delete functionality on products


Steps to test:
- In terminal, 
	- run nodemon app.js
	- run db/build-data.js to ensure faker json data is being created
	- run db/build-db.js to ensure sql table is being created
-  Make sure you can GET all products, GET one product, POST a product, PUT (edit) a product, and DELETE a product
